src/config/machines/example-grant-machine.js

```diff
@@ -67,7 +68,16 @@ export const actionImplementations = {
       delete newPageErrors[event.currentPageId];
       return newPageErrors;
     }
-  })
+  }),
+
+  /**
```

Comment:
XState has a dedicated `servcies` section for invoking long-running, asynchronous tasks (e.g., API calls, database queries, timers, etc.). Actions, on the other hand, are used for synchronous operations like logging, updating the context, or emitting additional events.

When to Use Services vs Actions
Use Services:

For long-running tasks that need asynchronous handling (e.g., API calls, timers).
When the task needs to determine the next state based on its success or failure.
Use Actions:

For synchronous operations like logging, updating the context, or emitting additional events.
When the operation doesn’t determine the next state.